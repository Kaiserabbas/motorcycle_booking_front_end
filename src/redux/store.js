import { configureStore } from '@reduxjs/toolkit';
import reserveReducer from './reserveSlice';
import motorcycleReducer from './motorcycleSlice';
import userReducer from './userSlice';

const store = configureStore({
  reducer: {
    reserve: reserveReducer,
    motorcycle: motorcycleReducer,
    user: userReducer,
  },
});

export default store;
