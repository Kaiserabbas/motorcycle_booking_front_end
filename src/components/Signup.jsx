import { useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { useNavigate } from 'react-router-dom';
import { cleanMessage, signupUser } from '../redux/userSlice';

const Signup = () => {
  const dispatch = useDispatch();
  const navegate = useNavigate();
  const [administrator, setAdministrator] = useState(false);
  const [newUser, setNewUser] = useState({
    name: '',
    email: '',
    password: '',
    admin: false,
  });

  const {
    information,
  } = useSelector((state) => state.user);

  return (
    <section className="loginContainer flexV">
      <div className="loginDiv">
        <form
          onSubmit={(el) => {
            el.preventDefault();
            dispatch(signupUser(newUser));
            el.target.reset();
          }}
          className="formContainer flexV"
        >
          <div className="flexV">
            <label htmlFor="name">
              Full Name
              <br />
              <input
                type="text"
                name="name"
                id="name"
                required
                onChange={(e) => {
                  if (e.target.value.length >= 2) {
                    setNewUser({
                      ...newUser,
                      name: e.target.value,
                    });
                  }
                }}
                minLength={2}
              />
            </label>
          </div>

          <div className="flexV">
            <label htmlFor="email">
              Email
              <br />
              <input
                type="email"
                name="email"
                id="email"
                required
                onChange={(e) => {
                  setNewUser({
                    ...newUser,
                    email: e.target.value,
                  });
                }}
              />
            </label>
          </div>

          <div className="flexV">
            <label htmlFor="password">
              Password
              <br />
              <input
                type="password"
                name="password"
                id="password"
                required
                onChange={(e) => {
                  setNewUser({ ...newUser, password: e.target.value });
                }}
              />
            </label>
          </div>

          <div className="flexV">
            <label htmlFor="admin">
              <h4>Administrator ?</h4>
              <input
                type="checkbox"
                name="admin"
                id="admin"
                onChange={() => {
                  setNewUser({ ...newUser, admin: !administrator });
                  setAdministrator(!administrator);
                }}
              />
            </label>
          </div>

          <div className="loginButtons flexH">
            <button
              id="signUp"
              type="submit"
            >
              Sign up
            </button>
            Or

            <button
              onClick={() => {
                dispatch(cleanMessage());
                navegate('/login');
              }}
              type="button"
            >
              Login
            </button>
          </div>
        </form>
        <div className="flexV infoContainer">
          {
            information && Array.isArray(information) ? information.map((sms) => (
              <p
                key={(1997 + Math.sin(Math.random() * 10))}
                className="error"
              >
                {sms}
              </p>
            )) : (
              information && (<p className="infoParagraph">{information}</p>)
            )
          }
        </div>
      </div>
    </section>
  );
};

export default Signup;
