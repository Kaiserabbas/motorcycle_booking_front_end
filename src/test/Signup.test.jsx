import '@testing-library/jest-dom/';
import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import { render } from '@testing-library/react';
import store from '../redux/store';
import Signup from '../components/Signup';

describe('Login component', () => {
  test('should render login page', () => {
    const component = render(
      <React.StrictMode>
        <Provider store={store}>
          <Router>
            <Signup />
          </Router>
        </Provider>
      </React.StrictMode>,
    );
    expect(component).toMatchSnapshot();
  });
});
