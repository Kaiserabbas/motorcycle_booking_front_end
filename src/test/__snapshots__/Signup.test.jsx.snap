// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component should render login page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="loginContainer flexV"
      >
        <div
          class="loginDiv"
        >
          <form
            class="formContainer flexV"
          >
            <div
              class="flexV"
            >
              <label
                for="name"
              >
                Full Name
                <br />
                <input
                  id="name"
                  minlength="2"
                  name="name"
                  required=""
                  type="text"
                />
              </label>
            </div>
            <div
              class="flexV"
            >
              <label
                for="email"
              >
                Email
                <br />
                <input
                  id="email"
                  name="email"
                  required=""
                  type="email"
                />
              </label>
            </div>
            <div
              class="flexV"
            >
              <label
                for="password"
              >
                Password
                <br />
                <input
                  id="password"
                  name="password"
                  required=""
                  type="password"
                />
              </label>
            </div>
            <div
              class="flexV"
            >
              <label
                for="admin"
              >
                <h4>
                  Administrator ?
                </h4>
                <input
                  id="admin"
                  name="admin"
                  type="checkbox"
                />
              </label>
            </div>
            <div
              class="loginButtons flexH"
            >
              <button
                id="signUp"
                type="submit"
              >
                Sign up
              </button>
              Or
              <button
                type="button"
              >
                Login
              </button>
            </div>
          </form>
          <div
            class="flexV infoContainer"
          />
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="loginContainer flexV"
    >
      <div
        class="loginDiv"
      >
        <form
          class="formContainer flexV"
        >
          <div
            class="flexV"
          >
            <label
              for="name"
            >
              Full Name
              <br />
              <input
                id="name"
                minlength="2"
                name="name"
                required=""
                type="text"
              />
            </label>
          </div>
          <div
            class="flexV"
          >
            <label
              for="email"
            >
              Email
              <br />
              <input
                id="email"
                name="email"
                required=""
                type="email"
              />
            </label>
          </div>
          <div
            class="flexV"
          >
            <label
              for="password"
            >
              Password
              <br />
              <input
                id="password"
                name="password"
                required=""
                type="password"
              />
            </label>
          </div>
          <div
            class="flexV"
          >
            <label
              for="admin"
            >
              <h4>
                Administrator ?
              </h4>
              <input
                id="admin"
                name="admin"
                type="checkbox"
              />
            </label>
          </div>
          <div
            class="loginButtons flexH"
          >
            <button
              id="signUp"
              type="submit"
            >
              Sign up
            </button>
            Or
            <button
              type="button"
            >
              Login
            </button>
          </div>
        </form>
        <div
          class="flexV infoContainer"
        />
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
